CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeEventManager.cs

```diff
@@ -575,13 +584,39 @@ private void PersistInboxMessages(List<object> messages)
             }
         }
 
+        private void OnMessagesUpdate()
+        {
+            CheckAndRemoveExpiredMessages();
```

Comment:
The expired messages should be removed from `OnInboxMessagesReceived` and `GetAllMessages` only.